src/status_im/protocol/handlers.cljs

```diff
@@ -548,6 +564,37 @@
 
 ;;GROUP
 
+(handlers/register-handler-fx
+  :group-chat-invite-received
+  (fn [{{:keys [current-public-key web3] :as db} :db}
+       [_ {:keys                                                    [from]
+           {:keys [group-id group-name contacts keypair timestamp]} :payload}]]
+    (let [{:keys [private public]} keypair]
+      (let [contacts' (keep (fn [ident]
+                              (when (not= ident current-public-key)
+                                {:identity ident})) contacts)
+            chat      {:chat-id     group-id
+                       :name        group-name
+                       :group-chat  true
+                       :group-admin from
+                       :public-key  public
+                       :private-key private
+                       :contacts    contacts'
+                       :added-to-at timestamp
+                       :timestamp   timestamp
+                       :is-active   true}
+            exists?   (chats/exists? group-id)]
+        (when (or (not exists?)
+                  (chats/new-update? timestamp group-id))
+          {:start-watching-group! {:web3     web3
+                                   :group-id group-id
+                                   :identity current-public-key
+                                   :keypair  keypair
+                                   :callback #(re-frame/dispatch [:incoming-message %1 %2])}
+           :dispatch              (if exists?
+                                    [:update-chat! chat]
+                                    [:add-chat group-id chat])})))))
```

Comment:
Instead of those dispatches, we should just call corresponding mode/event functions and include their effects, then no dispatch is necessary.

Reply:
I would like not to do this now, because in this PR I was focused only on jail handlers. Otherwise, it will become too big.